core/src/main/scala/org/apache/spark/h2o/converters/SparkDataFrameConverter.scala

```diff
@@ -138,7 +138,7 @@ private[h2o] object SparkDataFrameConverter extends Logging {
     con.closeChunks(localRowIdx)
 
     // Return H2O chunk index and number of rows in this chunk
-    (chunkIdx, con.numOfRows())
+    (chunkIdx, dataSize)
```

Comment:
@jakubhava Need to revert this line to fix tests on internal backend, since `dataSize` is `-1` for internal tests. if you don't mind, we can proceed with refactoring after the PR is merged.

Reply:
Yup, no worries, we'll clean this up after this PR is done, there will be more places we can simplify actually